src/workbookManager.ts

```diff
@@ -78,6 +80,169 @@ export class WorkbookManager {
+        if (workbookXmlString === undefined) {
+            throw new Error("Sheets were not found in template");
+        }
+        const newWorkbook = await this.updateWorkbookInitialData(workbookXmlString, tableData);
+        zip.file(workbookXmlPath, newWorkbook);
+    }
+
+    private async updatePivotTablesInitialData(tableXmlString: string, tableData: TableData) {
+        const parser: DOMParser = new DOMParser();
+        const serializer = new XMLSerializer();
+        const tableDoc: Document = parser.parseFromString(tableXmlString, "text/xml");
+
+        const tableColumns = tableDoc.getElementsByTagName("tableColumns")[0];
+        while (tableColumns.lastChild) {
+            tableColumns.removeChild(tableColumns.lastChild);
+        }
+        var columnIndex = 1;
+        tableData.columnNames.forEach(columnName => {
+            const tableColumn = tableDoc.createElementNS(tableDoc.documentElement.namespaceURI, "tableColumn");
+            tableColumn.setAttribute("id", columnIndex.toString());
+            tableColumn.setAttribute("uniqueName", columnIndex.toString());
+            tableColumn.setAttribute("name", columnName);
+            tableColumn.setAttribute("queryTableFieldId", columnIndex.toString());
+            tableColumns.appendChild(tableColumn);
+            columnIndex++;
+        });
+
+        tableColumns.setAttribute("count", tableData.columnNames.length.toString());
+        tableDoc.getElementsByTagName("table")[0].setAttribute("ref", `A1:${String.fromCharCode(tableData.columnNames.length + 64)}${(tableData.data.length + 1).toString()}`);
+        tableDoc.getElementsByTagName("autoFilter")[0].setAttribute("ref", `A1:${String.fromCharCode(tableData.columnNames.length + 64)}${(tableData.data.length + 1).toString()}`);
+        return serializer.serializeToString(tableDoc);
+    }
+
+     private async updateWorkbookInitialData(workbookXmlString: string, tableData: TableData) {
+        const newParser: DOMParser = new DOMParser();
+        const newSerializer = new XMLSerializer();
+        const workbookDoc: Document = newParser.parseFromString(workbookXmlString, "text/xml");
+        var definedName = workbookDoc.getElementsByTagName("definedName")[0];
+        definedName.textContent = "Query1!$A$1:$" + String.fromCharCode(tableData.columnNames.length + 64) + "$" + (tableData.data.length + 1).toString();
+        return newSerializer.serializeToString(workbookDoc);
+    }
+    
+    private async updateQueryTablesInitialData(queryTableXmlString: string, tableData: TableData) {
+        const parser: DOMParser = new DOMParser();
+        const serializer = new XMLSerializer();
+        const queryTableDoc: Document = parser.parseFromString(queryTableXmlString, "text/xml");
+        const queryTableFields = queryTableDoc.getElementsByTagName("queryTableFields")[0];
+        while (queryTableFields.lastChild) {
+            queryTableFields.removeChild(queryTableFields.lastChild);
+        }
```

Comment:
Take a look at this for a more up to date approach:
[Remove all child elements of a DOM node in JavaScript](https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript)](https://stackoverflow.com/a/65413839/2419693)